core/client/fs/src/main/java/alluxio/client/RetryHandlingMetaMasterClient.java

```diff
@@ -77,6 +83,28 @@ protected void afterConnect() {
     mClient = new MetaMasterClientService.Client(mProtocol);
   }
 
+  @Override
+  public synchronized ConfigCheckReport getConfigReport() throws IOException {
+    return retryRPC(() -> {
+      GetConfigReportTResponse response = mClient.getConfigReport(new GetConfigReportTOptions());
+
+      Map<Scope, List<InconsistentProperty>> wireErrors = new HashMap<>();
+      for (Map.Entry<alluxio.thrift.Scope, List<alluxio.thrift.InconsistentProperty>> entry :
+          response.getErrors().entrySet()) {
+        wireErrors.put(Scope.fromThrift(entry.getKey()), entry.getValue().stream()
+            .map(ThriftUtils::fromThrift).collect(Collectors.toList()));
+      }
+      Map<Scope, List<InconsistentProperty>> wireWarns = new HashMap<>();
+      for (Map.Entry<alluxio.thrift.Scope, List<alluxio.thrift.InconsistentProperty>> entry :
+          response.getWarns().entrySet()) {
+        wireWarns.put(Scope.fromThrift(entry.getKey()), entry.getValue().stream()
+            .map(ThriftUtils::fromThrift).collect(Collectors.toList()));
+      }
+      ConfigStatus status = ConfigStatus.fromThrift(response.getStatus());
+      return new ConfigCheckReport(wireErrors, wireWarns, status);
```

Comment:
Could we hide the thrift type -> wire type translation inside `ConfigCheckReport`? Then this could just be
```java
return retryRpc(() -> {
  return ConfigCheckReport.fromThrift(
      mClient.getConfigReport(new GetConfigReportTOptions()).getConfigReport());
}
```

Reply:
Changed, thanks